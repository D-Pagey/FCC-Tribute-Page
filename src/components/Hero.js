import React, { Component } from 'react';
import sharma from './sharma.jpg';

class Hero extends Component {
  render() {
    return (
      <div className="hero-container">
        <hr className="horizontal-rule" />
        <img src={sharma} className="hero-img" />
        <figcaption className="hero-img-caption">Chris Sharma's 2017 US Climbing Road Trip</figcaption>
      </div>
    )
  }
}

export default Hero;
