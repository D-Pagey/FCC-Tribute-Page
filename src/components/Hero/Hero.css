.hero-container {
    align-self: center;
    width: 92.5%; }

.horizontal-rule {
      align-self: center;
      border: 0;
      border-top: 1px solid;
      color: rgba(0, 0, 0, 0.1);
      margin: 1em 0 1em;
     }

.hero-img {
    border-radius: .5em;
    width: 100%; }

.hero-img-caption {
    color: gray;
    float: right;
    font-size: .75em;
    font-weight: 400;
    margin-top: .25em;
    padding-right: .35em; }
    @media screen and (min-width: 900px) {
      .hero-img-caption {
        font-size: 1.1rem; } }
    @media screen and (min-width: 2000px) {
      .hero-img-caption {
        font-size: 1.5em; } }
