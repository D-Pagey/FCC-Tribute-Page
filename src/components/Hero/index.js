import React from 'react';

import './index.css';
import sharma from './sharma.jpg';

export default function Hero() {
  return (
    <div className="hero-container">
      <hr className="horizontal-rule" />
      <img
        src={sharma}
        className="hero-img"
        alt="Chris Sharma climbing in the US."
      />
      <figcaption className="hero-img-caption">Chris Sharma&apos;s 2017 US Climbing
       Road Trip
      </figcaption>
    </div>
  );
}
