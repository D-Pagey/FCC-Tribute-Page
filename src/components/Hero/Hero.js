import React from 'react';

import './Hero.css';
import sharma from './sharma.jpg';

function Hero() {
  return (
    <div className='hero-container'>
      <hr className='horizontal-rule' />
      <img src={sharma} className='hero-img' alt='Chris Sharma climbing in the
       US.' />
      <figcaption className='hero-img-caption'>Chris Sharma's 2017 US Climbing
       Road Trip</figcaption>
    </div>
  )
}

export default Hero;
