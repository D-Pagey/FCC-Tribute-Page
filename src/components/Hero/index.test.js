import React from 'react';
import { shallow } from 'enzyme';

import Hero from '.';

it('render a Hero', () => {
  const wrapper = shallow(
    <Hero />,
  );
  expect(wrapper).toMatchSnapshot();
});
