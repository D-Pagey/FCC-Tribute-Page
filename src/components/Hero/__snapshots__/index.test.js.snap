// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Hero 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <hr
          className="horizontal-rule"
/>,
        <img
          alt="Chris Sharma climbing in the US."
          className="hero-img"
          src="sharma.jpg"
/>,
        <figcaption
          className="hero-img-caption"
>
          Chris Sharma's 2017 US Climbing Road Trip
</figcaption>,
      ],
      "className": "hero-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "horizontal-rule",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Chris Sharma climbing in the US.",
          "className": "hero-img",
          "src": "sharma.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Chris Sharma's 2017 US Climbing Road Trip",
          "className": "hero-img-caption",
        },
        "ref": null,
        "rendered": "Chris Sharma's 2017 US Climbing Road Trip",
        "type": "figcaption",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <hr
            className="horizontal-rule"
/>,
          <img
            alt="Chris Sharma climbing in the US."
            className="hero-img"
            src="sharma.jpg"
/>,
          <figcaption
            className="hero-img-caption"
>
            Chris Sharma's 2017 US Climbing Road Trip
</figcaption>,
        ],
        "className": "hero-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "horizontal-rule",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Chris Sharma climbing in the US.",
            "className": "hero-img",
            "src": "sharma.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Chris Sharma's 2017 US Climbing Road Trip",
            "className": "hero-img-caption",
          },
          "ref": null,
          "rendered": "Chris Sharma's 2017 US Climbing Road Trip",
          "type": "figcaption",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
